module/executiondatasync/optimistic_syncing/core.go

```diff
@@ -2,18 +2,288 @@ package pipeline
+			txResultErrMsgsRequester:      txResultErrMsgsRequester,
+			txResultErrMsgsRequestTimeout: txResultErrMsgsRequestTimeout,
+			indexer:                       indexerComponent,
+			persister:                     persisterComponent,
+			inmemRegisters:                inmemRegisters,
+			inmemEvents:                   inmemEvents,
+			inmemCollections:              inmemCollections,
+			inmemTransactions:             inmemTransactions,
+			inmemResults:                  inmemResults,
+			inmemTxResultErrMsgs:          inmemTxResultErrMsgs,
+		},
+		executionResult: executionResult,
+		header:          header,
+	}
+}
+
+// Download downloads execution data and transaction results error for the block
+// Expected errors:
+// - context.Canceled: if the provided context was canceled before completion
+// - context.DeadlineExceeded: if the provided context was canceled due to its deadline reached
+//
+// All other errors are unexpected and may indicate a bug or inconsistent state
+func (c *CoreImpl) Download(ctx context.Context) error {
+	c.log.Debug().Msg("downloading execution data")
+
+	g, gCtx := errgroup.WithContext(ctx)
+
+	var executionData *execution_data.BlockExecutionData
+	g.Go(func() error {
+		var err error
+		executionData, err = c.workingData.execDataRequester.RequestExecutionData(gCtx)
+		if err != nil {
+			return fmt.Errorf("failed to request execution data: %w", err)
+		}
+
+		return nil
+	})
+
+	var txResultErrMsgsData []flow.TransactionResultErrorMessage
+	g.Go(func() error {
+		timeoutCtx, cancel := context.WithTimeout(gCtx, c.workingData.txResultErrMsgsRequestTimeout)
+		defer cancel()
+
+		var err error
+		txResultErrMsgsData, err = c.workingData.txResultErrMsgsRequester.Request(timeoutCtx)
+		if err != nil {
+			if errors.Is(err, context.DeadlineExceeded) {
+				c.log.Debug().
+					Dur("timeout", c.workingData.txResultErrMsgsRequestTimeout).
+					Msg("transaction result error messages request timed out")
+				return nil
+			}
+
+			return fmt.Errorf("failed to request transaction result error messages data: %w", err)
+		}
+		return nil
+	})
+
+	if err := g.Wait(); err != nil {
+		// TODO: Improve error handling to allow partial success scenarios.
+		//
+		// Currently, this method uses errgroup with fail-fast semantics, meaning if either
+		// execution data OR transaction error messages download fails, the entire operation fails.
+		// However, these two types of data have different criticality levels:
+		//
+		// - executionData: CRITICAL - Required for block processing and indexing
+		// - txResultErrMsgsData: OPTIONAL - Nice-to-have
+		//
+		// Allow the download to succeed if execution data is retrieved
+		// successfully, even if transaction error messages fail or timeout. The system should:
+		//
+		// 1. Continue processing with empty/nil txResultErrMsgsData if that download fails
+		// 2. Log the failure appropriately
+		// 3. Allow operators to backfill missing error messages later
```

Comment:
you're already added the partial success scenario above. instead of this comment, can you add one where the error handling is done explaining the deadline exceeded error case
```suggestion
```